plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,360 @@
+        if available_name.group() in REBOOT_COMP:
+            reboot_uri.append(each["@odata.id"])
+            continue
+        prev_uri[each["Version"]] = each["@odata.id"]
+    if module.check_mode and (prev_uri or reboot_uri):
+        module.exit_json(msg=CHANGES_FOUND, changed=True)
+    elif (module.check_mode or not module.check_mode) and (not prev_uri and not reboot_uri):
+        module.exit_json(msg=NO_CHANGES_FOUND)
+    return list(prev_uri.values()), reboot_uri, update_uri
+
+
+def get_job_status(redfish_obj, module, job_ids, job_wait=True):
+    each_status, failed_count, js_job_msg = [], 0, ""
+    wait_timeout = module.params["reboot_timeout"]
+    for each in job_ids:
+        each_job_uri = MANAGER_JOB_ID_URI.format(each)
+        job_resp, js_job_msg = wait_for_redfish_job_complete(redfish_obj, each_job_uri, job_wait=job_wait,
+                                                             wait_timeout=wait_timeout)
+        if job_resp and js_job_msg:
+            module.exit_json(msg=JOB_WAIT_MSG.format(wait_timeout), job_status=[strip_substr_dict(job_resp.json_data)],
+                             changed=True)
+        job_status = job_resp.json_data
+        if job_status["JobState"] == "Failed":
+            failed_count += 1
+        strip_odata = strip_substr_dict(job_status)
+        each_status.append(strip_odata)
+    return each_status, failed_count
+
+
+def require_session(idrac, module):
+    session_id, token = "", None
+    payload = {'UserName': module.params["username"], 'Password': module.params["password"]}
+    path = SESSION_RESOURCE_COLLECTION["SESSION"]
+    resp = idrac.invoke_request('POST', path, data=payload, api_timeout=120)
+    if resp and resp.success:
+        session_id = resp.json_data.get("Id")
+        token = resp.headers.get('X-Auth-Token')
+    return session_id, token
+
+
+def wait_for_redfish_idrac_reset(module, redfish_obj, wait_time_sec, interval=30):
+    time.sleep(interval // 2)
+    msg = RESET_UNTRACK
+    wait = wait_time_sec
+    track_failed = True
+    resetting = False
+    while wait > 0 and track_failed:
+        try:
+            redfish_obj.invoke_request("GET", MANAGERS_URI, api_timeout=120)
+            msg = RESET_SUCCESS
+            track_failed = False
+            break
+        except HTTPError as err:
+            if err.getcode() == 401:
+                new_redfish_obj = Redfish(module.params, req_session=True)
+                sid, token = require_session(new_redfish_obj, module)
+                redfish_obj.session_id = sid
+                redfish_obj._headers.update({"X-Auth-Token": token})
+                track_failed = False
+                if not resetting:
+                    resetting = True
+                break
+            time.sleep(interval)
+            wait = wait - interval
+            resetting = True
+        except URLError:
+            time.sleep(interval)
+            wait = wait - interval
+            if not resetting:
```

Comment:
simple do resetting  = True, 

even if earlier resetting was False, it'll be overwritten and will become True, no need to do if statement to check earlier it was Fasle.

Reply:
I already observed the same, this is inside the loop for some cases the value is reset to False when it exits, so i added the specific condition